scripts/versions-replacer/src/main.rs

```diff
@@ -0,0 +1,66 @@
+use std::path::PathBuf;
+
+use argh::FromArgs;
+use color_eyre::{
+    eyre::{eyre, Context},
+    Result,
+};
+use regex::Regex;
+use walkdir::WalkDir;
+
+use versions_replacer::{
+    metadata::collect_versions_from_cargo_toml, replace::replace_versions_in_file,
+};
+
+#[derive(FromArgs)]
+/// Replace variables like '{{{{versions.fuels}}}}' with correct versions from Cargo.toml.
+/// Uses versions from '[workspace.members]' and '[workspace.metadata.versions-replacer.external-versions]'.
+struct VersionsReplacer {
+    /// path to directory with files containing variables
+    #[argh(positional)]
+    path: PathBuf,
+    /// path to Cargo.toml with versions
+    #[argh(option)]
+    manifest_path: PathBuf,
+    /// regex to filter filenames (example: "\.md$")
+    #[argh(option)]
+    filename_regex: Option<Regex>,
+}
+
+fn main() -> Result<()> {
+    let args: VersionsReplacer = argh::from_env();
+    let versions = collect_versions_from_cargo_toml(&args.manifest_path)?;
+
+    let mut total_replacements: Vec<usize> = Vec::new();
+
+    for entry in WalkDir::new(&args.path) {
+        let entry = entry.wrap_err("failed to get directory entry")?;
+
+        if entry.path().is_file() {
+            if let Some(filename_regex) = &args.filename_regex {
```

Comment:
Nit: the 'if' checks data that's not being updated in the loop. Would be nice if we factor it out. Or maybe we can omit the regex arg and hardcode it since we only use it on .md files?

Reply:
Thanks for the suggestion! I love it, though I don't want to rerequest 4 reviews again, so I'll skip it for now :)